assets/js/util/sort-by-property.js

```diff
@@ -17,19 +17,20 @@
  */
 
 /**
- * Sorts an object map based on numberic order/priority key.
+ * Sorts an array of objects based on numeric order/priority key.
  *
- * @param {Object} objectList  Object containg objects to be sorted
- * @param {string} priorityKey The object key to use for matching.
+ * @since n.e.x.t
  *
+ * @param {Array} arrayData  Array containing objects to be sorted
+ * @param {string} property The object key to use for matching.
  * @return {Array} An Array containing objects ordered by the priority key
  */
-export function sortObjectMapByKey( objectList, priorityKey ) {
-	return Object.values( objectList ).sort( ( objectA, objectB ) => {
-		if ( objectA[ priorityKey ] > objectB[ priorityKey ] ) {
+export function sortByProperty( arrayData, property ) {
```

Comment:
Is there a reason why we created a utility for this instead of using [`_.sortBy`](https://lodash.com/docs/4.17.15#sortBy)? It can be used the same way so I'm just curious if there was a reason or other decision to use our own implementation.

Reply:
We've been trying to use native JS implementations for simple things like this where it offers easy to use solutions so that we don't need to rely as much on `lodash`. But yeah, if we need to start splitting logic out into our own utility functions, I agree it becomes a bit less of a clear cut.

Reply:
Yeah that makes sense. There are several places in the older parts of the codebase where lodash `each`, `filter`,  or `map` are used that should probably use the native array methods as I'm not aware of any particular benefit there, other than maybe lodash offers better support with non-Arrays. I don't think we should be creating utilities though for something that is readily available (and likely much more thoroughly tested) via a dependency we already have and use.

Reply:
@aaemnnosttv @ryanwelcher I think overall we should try to slowly eliminate `lodash` since I question its benefits with what latest native array and object methods offer.
More specifically to these two functions though, I somewhat question whether we even need utilities for something simple like this. For example `sortByProperty` was previously accomplished in one location with `arr.sort( (a, b) => a[ prop ] - b[ prop ] )`.

While doing that inline and not using a utility requires a tiny bit more code (although not even that in every case), I'd argue that code might still be easier to follow - especially since as soon as we add utility functions like this to the mix, the code starts mixing native array functions and those more specific functions. That may be a personal preference, but I find that code harder to read than _only_ native array functions or _only_ utility functions.

I don't have a strong preference on keeping these utility functions or removing them again and keeping that functionality inline. We can probably keep them since they'll cover a lot of what we'd use `lodash` for. I think my main guidance at this point is let's try to not use `lodash` unless there's a significant benefit.